File: 438. Find All Anagrams in a String.java
public class Solution {
    public List<Integer> findAnagrams(String s, String p) {
        List<Integer> output = new ArrayList<Integer>();
        int[] pa = new int[26];
        int[] sa = new int[26];
        int pLen = p.length();
        boolean bool = true;
        
        for (int i = 0; i < pLen; i++) {
            pa[p.charAt(i) - 'a'] += 1;
        }
        
        for (int j = 0; j < s.length() - pLen + 1; j++) {
            String ss = s.substring(j, j + pLen);
            
            for (int k = 0; k < pLen; k++) {
                sa[ss.charAt(k) - 'a'] += 1;
            }
            for (int z = 0; z < 26; z++) {
                if (sa[z] != pa[z]) {
                    bool = false;
                    break;
                }
            }
            
            if (bool == true) {
                output.add(j);
            }
            
            bool = true;
            sa = new int[26];
        }
        return output;    
    }
}

File: 438. Find All Anagrams in a String.py
from collections import Counter

class Solution:
    def findAnagrams(self, s: str, p: str) -> List[int]:
        pLen = len(p)
        pCount = Counter(p)
        currCount = Counter(s[:pLen - 1])
        output = []
        
        for i in range(len(s) - pLen + 1):
            index = i + pLen - 1
            
            if s[index] in currCount:
                currCount[s[index]] += 1
            else:
                currCount[s[index]] = 1
            
            if pCount == currCount:
                output.append(i)
            
            currCount[s[i]] -= 1
            
            if currCount[s[i]] == 0:
                del currCount[s[i]]
        
        return output
